### PR TITLE
fix(package): use provided tsconfig when calling emitDts

### PR DESCRIPTION
**Pull request by quentinderoubaix**
Handles second task of https://github.com/sveltejs/language-tools/issues/2355

---

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### Edits

- [x] Please ensure that 'Allow edits from maintainers' is checked. PRs without this option may be closed.

### CODE REVIEW

File: packages/package/test/fixtures/svelte-3-types/tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
 	"compilerOptions": {
 		"target": "ESNext",
-		"module": "ESNext"
+		"module": "ESNext",
+		"verbatimModuleSyntax": true
```
Comment (dummdidumm) [inline packages/package/test/fixtures/svelte-3-types/tsconfig.json:5]: curious: why was this needed?

Comment (quentinderoubaix) [reply packages/package/test/fixtures/svelte-3-types/tsconfig.json:5]: Just to remove a warning while running the unit tests, smth like "tsconfig verbatimModuleSyntax is required when using Svelte files with lang=ts"

This is not related to the original goal of the PR, it can be reverted

**Review by dummdidumm — APPROVED** [review]
Thank you!
